### PR TITLE
Vep for sv

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist 

This PR adds vep to structural variant annotation workflow, and fixes a tiny issue in vcfanno (converts a string to a path using grrovy's file function). 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR> -stub`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -27,6 +27,10 @@ params {
     // Alignment
     aligner                    = 'bwamem2'
 
+    // Annotation
+    vep_genome                 = 'GRCh37'
```
Comment (jemten) [inline nextflow.config:31]: We should try to use the already given `--genome` parameter instead. Doesn't make sense to use align to one genome and annotate it with another. Might be that you have to parse that parameter but looking at the `genomes.config` you should be able to use it straight out of the box.

File: conf/modules.config [hunk 2]
```diff
@@ -202,6 +202,23 @@ process {
             enabled: false,
         ]
     }
+    withName: ENSEMBLVEP {
```
Comment (jemten) [inline conf/modules.config:205]: Maybe we should name it differently as we will use vep for SNV/INDELS and Mitochondria as well. Like ENSEMBLVEP_SV?

File: modules/local/ensemblvep/main.nf [hunk 3]
```diff
@@ -0,0 +1,52 @@
+process ENSEMBLVEP {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::ensembl-vep=104.3" : null)
```
Comment (jemten) [inline modules/local/ensemblvep/main.nf:5]: If at all possible we should have the same version of vep for conda as we have for the containers

File: main.nf [hunk 4]
```diff
@@ -26,6 +26,8 @@ params.svdb_query_dbs       = WorkflowMain.getGenomeAttribute(params, 'svdb_quer
 params.variant_catalog      = WorkflowMain.getGenomeAttribute(params, 'variant_catalog')
 params.vcfanno_resources    = WorkflowMain.getGenomeAttribute(params, 'vcfanno_resources')
 params.vcfanno_toml         = WorkflowMain.getGenomeAttribute(params, 'vcfanno_toml')
+params.vep_cache            = WorkflowMain.getGenomeAttribute(params, 'vep_cache')
+
```
Comment (jemten) [inline main.nf:30]:
```suggestion
```

File: modules/local/ensemblvep/main.nf [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        -o ${prefix}.ann.vcf \\
+        $args \\
+        --assembly $genome \\
+        --species $species \\
+        --cache \\
+        --cache_version $cache_version \\
+        --dir_cache $dir_cache \\
+        --fork $task.cpus \\
+        --vcf \\
+        --stats_file ${prefix}.summary.html
```
Comment (jemten) [inline modules/local/ensemblvep/main.nf:43]: We could add the option `--compress_output bgzip` as according to recommendations we should try to compress the output whenever possible. You would also have to change the out file format to `*.ann.vcf.gz`

**Review by jemten — COMMENTED** [review]
Nice work! 
Some minor comments

File: conf/modules.config (hunk 2 above)
Comment (ramprasadn) [reply conf/modules.config:205]: Yeah, good point! Will fix it in the next commit.

**Pull request merged**